btf/strings.go

```diff
@@ -203,19 +197,14 @@ func (stb *stringTableBuilder) Length() int {
 	return int(stb.length)
 }
 
-// Marshal a string table into its binary representation.
-func (stb *stringTableBuilder) Marshal() []byte {
-	buf := make([]byte, stb.Length())
-	stb.MarshalBuffer(buf)
-	return buf
-}
-
-// Marshal a string table into a pre-allocated buffer.
-//
-// The buffer must be at least of size Length().
-func (stb *stringTableBuilder) MarshalBuffer(buf []byte) {
+// AppendEncoded appends the string table to the end of the provided buffer.
+func (stb *stringTableBuilder) AppendEncoded(buf []byte) []byte {
+	n := len(buf)
+	buf = append(buf, make([]byte, stb.Length())...)
+	strings := buf[n:]
 	for str, offset := range stb.strings {
-		n := copy(buf[offset:], str)
-		buf[offset+uint32(n)] = 0
+		n := copy(strings[offset:], str)
+		strings[offset+uint32(n)] = 0
```

Comment:
Is this necessary? I'd expect this slot to be zero due to the append above, but I assume you added it for a reason. I guess `bytes.Buffer.Bytes()` synthesizes a slice header pointing to a backing array that may or may not have gone through the pool, and append() doesn't implicitly zero if it doesn't realloc? That's an interesting edge case, and we should probably document that somewhere around here..

Reply:
No, I just kept it from the previous code. I'll remove it.